library/gcm.c

```diff
@@ -417,6 +417,7 @@ static int gcm_mask(mbedtls_gcm_context *ctx,
     return 0;
 }
 
+MBEDTLS_OPTIMIZE_ALWAYS
```

Comment:
What about if someone wants to build without -O2 to be able to debug?

Reply:
This is only enabled if the user passes -Os

Reply:
I've improved the docs here.